src/sync/partial_sync.hpp

```diff
@@ -107,6 +108,7 @@ RowExpr subscribe_blocking(Results const&, util::Optional<std::string> name);
 /// The operation is performed asynchronously. Completion will be indicated by the
 /// `Subscription` transitioning to the `Invalidated` state.
 void unsubscribe(Subscription&);
+void unsubscribe(Object&&);
```

Comment:
Perhaps add a comment here that it needs to be a managed RealmObject of the type `__ResultSets`... at least I was a bit confused about it when just glancing over the code.